### PR TITLE
Add FAQ for cluster-api and cdk8s

### LINKED ISSUE #223: Add FAQ item on CDK/cdk8s
Our [FAQ](https://aws.github.io/aws-controllers-k8s/community/faq/) should have an item that explains how ACK relates to CDK and cdk8s.

### PR DESCRIPTION
**Pull request by rothgar**
fixes #223 

Add two additional FAQ questions/answers.

### CODE REVIEW

File: docs/contents/community/faq.md [hunk 1]
```diff
@@ -12,11 +12,46 @@
 
     We appreciate input and advice concerning features that are currently covered by the service broker only, for example bind/unbind or cataloging and looking forward to learn from the community how they are using service broker so that we can take this into account.
 
+## Cluster API
+
+!!! question "Question"
+    Does ACK replace [cluster api](https://github.com/kubernetes-sigs/cluster-api)?
```
Comment (jaypipes) [inline docs/contents/community/faq.md:18]: Should this be made more explicit as "Does the planned ACK service controller for EKS replace Kubernetes Cluster API?"

File: docs/contents/community/faq.md [hunk 2]
```diff
@@ -12,11 +12,46 @@
...
+
+!!! quote "Answer"
+    Cluster API does a lot of really cool things and is designed to be a generic way to create Kubernetes clusters that run anywhere.
```
Comment (jaypipes) [inline docs/contents/community/faq.md:21]: I would start this answer with a specific: "No, the ACK service controller for EKS does not replace Kubernetes Cluster API."

and then continue with another paragraph about "Cluster API does a lot of really cool things..."

File: docs/contents/community/faq.md [hunk 3]
```diff
@@ -12,11 +12,46 @@
...
+    It makes some different design decisions with that goal in mind.
+    Some differences include:
+
+    - Cluster API is treated as your source of truth for all infrastructure.
+    This means things like the cluster autoscaler need to be configured to use cluster api instead of AWS cloud provider.
+    ACK provisions clusters using AWS as the source of truth the same way any EKS cluster does.
+    - Generic Kubernetes clusters rely on running more services in the cluster and not services from AWS.
+    Things like metrics and logging will likely need to run inside Kubernetes instead of using services like CloudWatch.
+    - IAM permission for Cluster-API Provider AWS (CAPA) need to be more broad than the EKS controller because CAPA is responsible for provisioning everything needed for the cluster (VPC, gateway, etc).
```
Comment (jaypipes) [inline docs/contents/community/faq.md:30]: s/the EKS controller/the ACK service controller for EKS/g

File: docs/contents/community/faq.md [hunk 4]
```diff
@@ -12,11 +12,46 @@
...
+
+## cdk8s
+
+!!! question "Question"
+    How does ACK relate to [cdk8s](https://cdk8s.io/)?
+
+!!! quote "Answer"
+    cdk8s is an open-source software development framework for defining Kubernetes applications and reusable abstractions using familiar programming languages and rich object-oriented APIs.
+    You can use cdk8s to create any resource inside a Kubernetes cluster.
+    This includes CustomResourceDefinition (CRD) resources.
```
Comment (jaypipes) [inline docs/contents/community/faq.md:44]: 1) Should be Custom Resources (CRs)

2) Maybe linkify "Custom Resources" to point here: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/

File: docs/contents/community/faq.md [hunk 5]
```diff
@@ -12,11 +12,46 @@
...
+    
+    All of the ACK controllers watch for specific CRDs and you can generate those resources using cdk8s or any Kubernetes tooling.
```
Comment (jaypipes) [inline docs/contents/community/faq.md:46]: s/CRDs/CRs/

File: docs/contents/community/faq.md [hunk 6]
```diff
@@ -12,11 +12,46 @@
...
+    The two projects compliment eachother.
```
Comment (jaypipes) [inline docs/contents/community/faq.md:47]: s/compliment/complement/

File: docs/contents/community/faq.md [hunk 7]
```diff
@@ -12,11 +12,46 @@
...
 ## Contributing
 
 !!! question "Question"
     Where and how can I help?
 
 !!! quote "Answer"
     Excellent question and we're super excited that you're interested in ACK.
-    For now, if you're a developer, you can check out the [mvp](https://github.com/aws/aws-controllers-k8s/tree/mvp) branch and try out the code generation.
\ No newline at end of file
+    For now, if you're a developer, you can check out the [contributor docs](../../dev-docs/overview/).
```
Comment (jaypipes) [inline docs/contents/community/faq.md:57]: :+1: nice catch!

File: docs/contents/community/faq.md (hunk 2 above)
Comment (mhausenblas) [reply docs/contents/community/faq.md:21]: Yep, what Jay said. We should strive to always clearly say Yes or No in the first sentence (if it's a binary question).

File: docs/contents/community/faq.md [hunk 8]
```diff
@@ -12,11 +12,46 @@
...
+    Does ACK replace [cluster api](https://github.com/kubernetes-sigs/cluster-api)?
+
+!!! quote "Answer"
+    Cluster API does a lot of really cool things and is designed to be a generic way to create Kubernetes clusters that run anywhere.
+    It makes some different design decisions with that goal in mind.
+    Some differences include:
+
+    - Cluster API is treated as your source of truth for all infrastructure.
+    This means things like the cluster autoscaler need to be configured to use cluster api instead of AWS cloud provider.
+    ACK provisions clusters using AWS as the source of truth the same way any EKS cluster does.
```
[the remark this answers is not in the document]
Comment (mhausenblas) [reply docs/contents/community/faq.md:27]: Yeah, I'd just say: they are trying to do different things and then explain that there is little to no overlap.

**Review by mhausenblas — CHANGES_REQUESTED** [review]
Looking really good, thanks a million Justin! Some (minor) things to address and then we're good to 🚢

**Comment by rothgar** [comment]
Thanks all, I think I addressed all the feedback

File: docs/contents/community/faq.md [hunk 9]
```diff
@@ -54,4 +54,4 @@
 
 !!! quote "Answer"
     Excellent question and we're super excited that you're interested in ACK.
-    For now, if you're a developer, you can check out the [contributor docs](../../dev-docs/overview/).
\ No newline at end of file
+    For now, if you're a developer, you can check out the [contributor docs](../contents/dev-docs/overview/).
```
Comment (mhausenblas) [inline docs/contents/community/faq.md:57]: Nope, the previous link (`../../dev-docs/overview/`) is the correct one. Please always check with `mkdocs serve` locally. Side note to myself: we really should make link checker part of the CI.

**Review by mhausenblas — CHANGES_REQUESTED** [review]

File: docs/contents/community/faq.md (hunk 9 above)
Comment (rothgar) [reply docs/contents/community/faq.md]: I was previewing it locally which is why I was surprised the 2nd comment said the link was broken. I realized I fix that but forgot to save/commit/push so when I saw the comment I unfixed it and push it broken.

looks like with `strict: true` in the mkdocks.yaml we could error if there were warnings for things like broken links but that may cause other friction for small warnings.

Comment (mhausenblas) [reply docs/contents/community/faq.md]: Gotcha, thanks for the clarification and that's a great suggestion. Will play around with it and see if we can use it.

**Comment by joebowbeer** [comment]
#223 was also asking about CDK for the inquiring knaves. Is there anyway to relate ACK to CDK?

**Pull request merged**